app/src/main/java/com/example/triptracker/view/map/MapOverview.kt

```diff
@@ -0,0 +1,176 @@
+package com.example.triptracker.view.map
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.util.Log
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.Icon
+import androidx.compose.material.IconButton
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Brush
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.font.FontFamily
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.example.triptracker.R
+import com.example.triptracker.navigation.AllowLocationPermission
+import com.example.triptracker.navigation.checkForLocationPermission
+import com.example.triptracker.navigation.getCurrentLocation
+import com.example.triptracker.viewmodel.MapViewModel
+import com.google.android.gms.maps.model.CameraPosition
+import com.google.android.gms.maps.model.LatLng
+import com.google.maps.android.compose.GoogleMap
+import com.google.maps.android.compose.MapProperties
+import com.google.maps.android.compose.MapType
+import com.google.maps.android.compose.MapUiSettings
+import com.google.maps.android.compose.rememberCameraPositionState
+
+@SuppressLint("UnusedMaterialScaffoldPaddingParameter")
+@Composable
+/**
+ * Composable displaying the map overview with all the paths and markers of trips that are around
+ * the user's location. Needs the context of the app in order to ask for location permission and
+ * access location.
+ */
+fun MapOverview(
+    mapViewModel: MapViewModel = MapViewModel(),
+    context: Context,
```

Comment:
Maybe add comments about those in the description of the MapOverview function with the "@ param" marker ?